.ci/test_r_package.sh

```diff
@@ -236,12 +236,10 @@ if [[ $used_correct_r_version -ne 1 ]]; then
     echo "Unexpected R version was used. Expected '${R_VERSION}'."
     exit 1
 fi
-
 if [[ $R_BUILD_TYPE == "cmake" ]]; then
-    passed_correct_r_version_to_cmake=$(
```

Comment:
what is the purpose for these changes? Can you please explain them?

Providing the text of the exact error(s) from `shellcheck` that you're attempting to address would be helpful.

Reply:
In testing.sh line 241:
    passed_correct_r_version_to_cmake=$(
    ^-- SC2034 (warning): passed_correct_r_version_to_cmake appears unused. Verify use (or export if used externally).

And after declaring the variable as export this was the error
In testing.sh line 241:
    export passed_correct_r_version_to_cmake=$(
           ^-- SC2155 (warning): Declare and assign separately to avoid masking return values.

So then i declared passed_correct_r_version_to_cmake seperately and then made the count variable and initialized passed_correct_r_version_to_cmake to count and check for the version

Reply:
Thanks. Note that I've edited your post, to make the difference between code, output from code, and your own words clearer. You may want to review https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax.

---

This fix does not look correct to me. It looks like the actual issue is that this check is storing a value in a variable called `passed_correct_r_version_to_cmake`, but then the `if` statement below is referencing a variable called `used_correct_r_verrsion`.

Please revert your changes in this part of the code, and instead change this line:

https://github.com/microsoft/LightGBM/blob/fd8f79944955cf83fcefb9ed598b9e0bcd07ad09/.ci/test_r_package.sh#L245

to use `passed_correct_r_version_to_cmake`

Reply:
Changed the variable name. Thank you for the link to the github docs

Reply:
I think you've misunderstood. Here in this part of the code, the variable name should be changed **and nothing else should**.

All of the other changes you've made (like marking `passed_correct_r_version_to_cmake` for export), should be reverted. Please revert them.

Reply:
Did the change. Sorry for misunderstanding